fixql-core/src/main/scala/com/iterable/graphql/FromGraphQLJava.scala

```diff
@@ -0,0 +1,253 @@
+package com.iterable.graphql
+
+import java.util.concurrent.CompletableFuture
+
+import com.iterable.graphql.compiler.Schema
+import graphql.ExecutionInput
+import graphql.Scalars
+import graphql.execution.ExecutionContext
+import graphql.execution.ExecutionStepInfo
+import graphql.execution.FetchedValue
+import graphql.execution.MergedField
+import graphql.execution.ValuesResolver
+import graphql.execution.nextgen.ExecutionStrategy
+import graphql.execution.nextgen.FetchedValueAnalysis
+import graphql.execution.nextgen.FieldSubSelection
+import graphql.execution.nextgen.result.ExecutionResultNode
+import graphql.execution.nextgen.result.LeafExecutionResultNode
+import graphql.execution.nextgen.result.RootExecutionResultNode
+import graphql.introspection.Introspection
+import graphql.language.{Field => JField}
+import graphql.schema.DataFetchingFieldSelectionSet
+import graphql.schema.DataFetchingFieldSelectionSetImpl
+import graphql.schema.GraphQLTypeUtil
+import graphql.schema.idl.EchoingWiringFactory
+import graphql.schema.idl.SchemaGenerator
+import graphql.schema.GraphQLSchema
+import graphql.schema.idl.RuntimeWiring
+import graphql.schema.idl.SchemaParser
+import io.circe.Json
+import io.circe.JsonNumber
+import io.circe.JsonObject
+import play.api.libs.json.{Json => PlayJson}
+import play.api.libs.json.JsArray
+import play.api.libs.json.JsBoolean
+import play.api.libs.json.JsNull
+import play.api.libs.json.JsNumber
+import play.api.libs.json.JsString
+import play.api.libs.json.JsValue
+
+import scala.concurrent.Future
+import scala.collection.JavaConverters.iterableAsScalaIterableConverter
+import scala.collection.JavaConverters.mapAsScalaMapConverter
+import scala.collection.JavaConverters.mapAsJavaMapConverter
+import scala.util.Try
+
+/**
+  * Document and schema parsing using the GraphQL Java library.
+  */
+object FromGraphQLJava {
+
+  /** Parse and validate documents (queries and mutations) using the GraphQL Java library.
+    * This requires a GraphQL Java GraphQLSchema. Returns our Query type.
+    *
+    * Overall flow looks as follows:
+    * - Build a GraphQL Java GraphQLSchema whether from SDL or programmatically.
+    * - Use GraphQL Java to parse the document (i.e. query) into Java AST
+    * - Use GraphQL Java to resolve variables etc.
+    * - Convert to Scala AST by traversing Java using selection-set peeking. This is what the code below does.
+    * - Compile Scala AST into DBIO
+    * - Run DBIO
+    *
+    * This method uses "field selection sets" from GraphQL-Java to extract a Field tree for the query:
+    * https://www.graphql-java.com/documentation/v12/fieldselection/
+    */
+  def parseAndValidateQuery(graphQLSchema: GraphQLSchema, query: String, variables: Json): Try[Query[Field.Fixed]] = {
```

Comment:
the code below may look rather painful. Basically we half-execute the query using GraphQL-Java, tricking it into giving us the tree of query fields, that we then convert into our AST.